conflation/trace_fetching/mapillary.py

```diff
@@ -198,181 +206,335 @@ def make_trace_data_requests(
-            #  in, that way we won't ever have too many files in one dir
-            print(
-                "WARNING: {} bbox sections will be generated and a .pickle file will be created for all of them, "
-                "violating the MAX_FILES_IN_DIR={}".format(num_files, MAX_FILES_IN_DIR)
-            )
-        else:
-            print("{} bbox sections will be generated...".format(num_files))
-        """
         bbox_sections = []
 
-        tile1 = get_tile_from_lon_lat(min_long, min_lat, zoom)
-        tile2 = get_tile_from_lon_lat(max_long, max_lat, zoom)
+        tile1 = get_tile_from_lon_lat(min_lon, max_lat, COVERAGE_ZOOM)
+        tile2 = get_tile_from_lon_lat(max_lon, min_lat, COVERAGE_ZOOM)
 
         start_x, end_x = min(tile1[0], tile2[0]), max(tile1[0], tile2[0])
         start_y, end_y = min(tile1[1], tile2[1]), max(tile1[1], tile2[1])
 
+        print(
+            "Searching through zoom=5 tiles from ({}, {}) to ({}, {})".format(
+                start_x, start_y, end_x, end_y
+            )
+        )
         for x in range(start_x, end_x + 1):
             for y in range(start_y, end_y + 1):
-                print(x, y, zoom)
-                # TODO
-
-        """
-        prev_long = min_long
-        while prev_long < max_long:
-            cur_long = min(prev_long + section_size, max_long)
-            prev_lat = min_lat
-            while prev_lat < max_lat:
-                cur_lat = min(prev_lat + section_size, max_lat)
-
-                # Convert the long / lat bbox bounds to a string that the trace source API can understand (using the
-                # given lambda)
-                bbox_str = to_bbox_str(prev_long, prev_lat, cur_long, cur_lat)
-
-                # The file on disk where we will store trace data
-                trace_filename = os.path.join(
-                    traces_dir, util.get_sha1_truncated_id(bbox_str) + ".pickle"
+                # Create a dir to store trace data for this zoom 5 tile
+                zoom_5_dir = os.path.join(
+                    traces_dir, "_".join([str(COVERAGE_ZOOM), str(x), str(y)])
+                )
+                if not os.path.exists(zoom_5_dir):
+                    os.makedirs(zoom_5_dir)
+
+                # At 14, the top left corner tile (i.e. pixel (0, 0) at zoom 5 tile)
+                base_x_zoom_14 = x * 2 ** (BBOX_SECTION_ZOOM - COVERAGE_ZOOM)
+                base_y_zoom_14 = y * 2 ** (BBOX_SECTION_ZOOM - COVERAGE_ZOOM)
+
+                resp = session_.get(
+                    COVERAGE_TILES_URL.format(COVERAGE_ZOOM, x, y, access_token_)
                 )
 
-                bbox_sections.append((bbox_str, trace_filename))
-                prev_lat += section_size
-            prev_long += section_size
-        """
+                tile_pb = vector_tile_pb2.Tile()
+                tile_pb.ParseFromString(resp.content)
+
+                add_z14_tiles_from_coverage_tile_to_bbox_sections(
```

Comment:
nit: could we return the list of new sections and then extend the existing list here? it feels a bit weird to have the method modify the passed in arg. no worries if this is a stupid idea

Reply:
At first I thought "yep that's a good idea", then when I went in to make the change I thought "ah I see why I did it the other way", but ended up changing it anyways since at least it reduces the number of arguments in this method.